### PR TITLE
build: convert build scripts to Kotlin and target Java 17

### LINKED ISSUE #71: Migrate buildscript to Kotlin

### PR DESCRIPTION
**Pull request by broccolai**
closes #71 and #72

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import net.kyori.indra.repository.sonatypeSnapshots
+
+plugins {
+  val indraVersion = "2.0.6"
```
Comment (Proximyst) [inline build.gradle.kts:4]: Might as well just use the aliases with version catalogs?

**Comment by broccolai** [comment]
Updated the Dockerfile too but I am by no means a docker expert so that should definitely be reviewed :')

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
-ARG JAVA_VERSION=16
+ARG JAVA_VERSION=17
 ARG JVM_FLAVOR=hotspot
 
-FROM adoptopenjdk:${JAVA_VERSION}-jdk-${JVM_FLAVOR} AS builder
+FROM openjdk:${JAVA_VERSION}-slim AS builder
```
Comment (yannicklamprecht) [inline Dockerfile:4]: You likely want the JDK for compiling the project.

`openjdk:17-jdk-slim`


```suggestion
FROM openjdk:${JAVA_VERSION}-jdk-slim AS builder
```

**Review by yannicklamprecht — CHANGES_REQUESTED** [review]

**Review by yannicklamprecht — APPROVED** [review]
LGTM

Probably a follow up task would be to check if a general dependency can be upgrade.
Just as a recurring task or by using dependabot.

**Review by kashike — APPROVED** [review]

**Pull request merged**